### ISSUE TITLE
Infinite watch stream loop

### ISSUE THREAD

**Issue by oir** [tomplus/kubernetes_asyncio#329]
Hi! The below snippet works just fine on version 30.1.1 when I invoke on a pod in "Succeeded" state:

```python
import asyncio

from kubernetes_asyncio import client, watch, config
from kubernetes_asyncio.client.api_client import ApiClient

if __name__ == "__main__":
    async def main():
        pod = "..."
        namespace = "..."
        await config.load_kube_config()

        async with ApiClient() as api:
            watcher = watch.Watch()
            async for log in watcher.stream(client.CoreV1Api(api).read_namespaced_pod_log, pod, namespace):
                print(log, end="")

    asyncio.run(main())
```
the loop ends and execution stops after the logs finish.

However same snippet keeps looping forever on version 30.3.0. Once the logs finish, it starts over and repeats.

**Comment by tomplus** [comment]
Thanks for reporting this issue and sorry for the inconvenience.

I try to fix it but it looks like the Watch() tries to stream "events" forever (fixed in 30.3.0) and I'm not sure if it's possible to differentiate timeout and end of logs. I'll dig into it...

Is there any reason why you use Watch() in this scenario? You can try https://github.com/tomplus/kubernetes_asyncio/blob/master/examples/tail.py#L41 this approach, should work in you case.

**Comment by tomplus** [comment]
It has been fixed in version 30.3.1

**Comment by oir** [comment]
That was fast, thank you! I was simply unaware that there was any other option, I confirm your snippet works just fine as well, in 30.3.1. 👍
